libvast/src/evaluate.cpp

```diff
@@ -27,60 +27,56 @@ namespace vast {
 
 namespace {
 
-template <class LhsView, class RhsView>
+template <class LhsView, class Rhs>
 inline constexpr auto requires_stdcmp
-  = std::is_integral_v<LhsView> && std::is_integral_v<RhsView>
-    && !std::is_same_v<LhsView, RhsView>
-    && !detail::is_any_v<bool, LhsView, RhsView>;
+  = std::is_integral_v<LhsView> && std::is_integral_v<Rhs>
+    && !std::is_same_v<LhsView, Rhs>
+    && !detail::is_any_v<bool, LhsView, Rhs>;
 
 template <relational_operator Op>
 struct cell_evaluator;
 
 template <>
 struct cell_evaluator<relational_operator::equal> {
-  static bool evaluate(auto, auto) noexcept {
+  static bool evaluate(auto, const auto&) noexcept {
```

Comment:
> **Note**
> We actually always have a `const data&` on the right-hand side here, so passing in a view was unnecessary, and now is actively harmful for patterns, where we don't want to instantiate the regex again.

Reply:
Can this be extracted into a separate `re2`-less pull request, just for the 3.0 release?